### ISSUE TITLE
DITA 1.3 catalog.xml points to non existing resources

### ISSUE THREAD

**Issue by raducoravu** [dita-ot/dita-ot#2266]
The XML catalog:

plugins\org.oasis-open.dita.v1_3\dtd\base\catalog.xml
A bunch of references to ".ent" files like:

        <public publicId="-//OASIS//ENTITIES DITA 1.3 Map//EN" uri="dtd/map.ent"/>

or the references for "topic.ent" point to non existing resources.

**Comment by raducoravu** [comment]
Same for this catalog:

org.oasis-open.dita.v1_3\dtd\machineryIndustry\catalog.xml

**Comment by raducoravu** [comment]
This catalog also references non-existing resources:

       plugins\org.oasis-open.dita.v1_3\schema\learning\catalog.xml

**Comment by raducoravu** [comment]
And a couple more catalogs, the full list:

            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\dtd\base\dtd\map.ent does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/dtd/base/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\dtd\base\dtd\map.ent does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/dtd/base/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\dtd\base\dtd\topic.ent does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/dtd/base/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\dtd\base\dtd\topic.ent does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/dtd/base/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\dtd\machineryIndustry\dtd\machineryTaskbodyConstraint.ent does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/dtd/machineryIndustry/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\dtd\machineryIndustry\dtd\machineryTaskbodyConstraint.ent does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/dtd/machineryIndustry/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\schema\learning\xsd\learningAggregationsTopicrefConstraintGrp.xsd does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/schema/learning/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\schema\learning\xsd\learningAggregationsTopicrefConstraintGrp.xsd does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/schema/learning/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\schema\learning\xsd\learningAggregationsTopicrefConstraintGrp.xsd does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/schema/learning/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\schema\learning\xsd\learningAggregationsTopicrefConstraintGrp.xsd does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/schema/learning/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\schema\technicalContent\ditabaseMod.xsd does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/schema/technicalContent/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\schema-url\learning\xsd\learningAggregationsTopicrefConstraintGrp.xsd does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/schema-url/learning/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\schema-url\learning\xsd\learningAggregationsTopicrefConstraintGrp.xsd does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/schema-url/learning/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\schema-url\learning\xsd\learningAggregationsTopicrefConstraintGrp.xsd does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/schema-url/learning/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\schema-url\learning\xsd\learningAggregationsTopicrefConstraintGrp.xsd does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/schema-url/learning/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\schema-url\technicalContent\glossgroup.xsd does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/schema-url/technicalContent/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\schema-url\technicalContent\glossgroup.xsd does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/schema-url/technicalContent/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\schema-url\technicalContent\glossgroup.xsd does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/schema-url/technicalContent/catalog.xml
            The file:D:\projects\eXml\frameworks\dita\DITA-OT2.x\plugins\org.oasis-open.dita.v1_3\schema-url\technicalContent\glossgroup.xsd does not exist from file:/D:/projects/eXml/frameworks/dita/DITA-OT2.x/plugins/org.oasis-open.dita.v1_3/schema-url/technicalContent/catalog.xml

**Comment by robander** [comment]
These catalogs are coming un-modified from the OASIS distribution; I'm guessing that the RNG->DTD conversion tool inserted entries for these, but you're right that the files do not exist. Probably makes sense for us to remove them before OASIS gets around to it, but I'll send a note to OASIS so it's added to the errata.

**Comment by robander** [comment]
Follow-up: I sent email to OASIS, then noticed that one of the entries in the catalog is wrong (rather than extra).

This:
```xml
   <public publicId="-//OASIS//ENTITIES DITA 1.3 Topic//EN" uri="dtd/topic.ent"/>
   <public publicId="-//OASIS//ENTITIES DITA Topic//EN" uri="dtd/topic.ent"/>
```

should be
```xml
   <public publicId="-//OASIS//ENTITIES DITA 1.3 Topic Definitions//EN" uri="dtd/topicDefn.ent"/>
   <public publicId="-//OASIS//ENTITIES DITA Topic Definitions//EN" uri="dtd/topicDefn.ent"/>
```

**Comment by drmacro** [comment]
Yes, definitely wrong. However, in normal practice the public ID will never be used be topicDefn.ent is only ever used (and should only ever be used) from topic.mod, which is probably why I missed this in my testing.

I'll go over the current catalog as soon as I'm able.

**Comment by robander** [comment]
I'm putting together a pull request for this - you / we can probably use that as a basis for updating the catalogs with Errata.

[And yes, having the correct system ID in `topic.mod` makes this work, and is presumably why nobody has noticed.]

**Comment by robander** [comment]
Ah, one more error, but I'm less certain how to fix this:
```xml
      <system systemId="ditabaseMod.xsd" uri="ditabaseMod.xsd"/>
      <system systemId="urn:oasis:names:tc:dita:xsd:ditabaseMod.xsd"
              uri="xsd/ditabaseMod.xsd"/>
```

Should that first entry exist? I'm pretty sure the answer is "no" but want to verify.

**Comment by robander** [comment]
And the last error:
```xml
      <!-- DITA 1.2 versions of the URNs for glossgroup shell. -->
      <system systemId="urn:oasis:names:tc:dita:xsd:glossarygroup.xsd" uri="glossgroup.xsd"/>
      <system systemId="urn:oasis:names:tc:dita:xsd:glossarygroup.xsd:1.2" uri="glossgroup.xsd"/>
```

In this case - I think the first one should be updated to reference the actual 1.3 version of the file `xsd/glossgroup.xsd`, and the second one should probably be removed, as this is not a 1.2 version of the file. (In any context, toolkit or not, if I specifically reference the 1.2 URI then I don't want this catalog taking over and getting me the 1.3 version.)

**Comment by easirois** [comment]
@robander the ditabaseMod.xsd without the URN is carry over from DITA 1.2.  I can't recall exactly why is was there.  It may have been there to address some issue with the OASIS catalog parser.  The fist one can disappear.  I would expect the 1.x can disappear as well and update the 1.2 to 1.3.

**Comment by robander** [comment]
Thanks @easirois I'll have a pull request in here today.

**Comment by robander** [comment]
@easirois and @drmacro if you have a moment to review that commit and make sure it all looks good, I'd appreciate it.

**Comment by robander** [comment]
Fix integrated for 2.2.5.
